src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -155,7 +156,14 @@ private void addBinaryToPath(String hdfsBinaryPath) throws IOException {
     BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
     bufferedWriter.write(scriptContent);
     bufferedWriter.close();
-    Runtime.getRuntime().exec("chmod a+x " + pathEnvVarLocation);
+    ProcessBuilder processBuilder = new ProcessBuilder("chmod", "a+x", pathEnvVarLocation);
+    Process process = processBuilder.start();
+    int exitCode = process.waitFor();
+    if (exitCode != 0) {
+      log.fatal("Error creating the symbolic link to hdfs binary."
+          + "Failure running chmod a+x " + pathEnvVarLocation);
+      System.exit(1);
```

Comment:
So now createSymbolicLink and addBinaryToPath are the only places that the executor exits on its own. Good progress, but are even these errors really fatal? Will the scheduler be able to recover?